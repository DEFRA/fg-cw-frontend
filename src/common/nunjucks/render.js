import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
