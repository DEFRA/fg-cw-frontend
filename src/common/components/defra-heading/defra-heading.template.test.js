import { describe, expect, test } from "vitest";
import { render } from "../../nunjucks/render.js";

describe("defra-heading", () => {
  test("renders", () => {
    const component = render("defra-heading", {
      serviceName: "Manage rural grant applications",
    });

    expect(component).toMatchSnapshot();
  });
});
