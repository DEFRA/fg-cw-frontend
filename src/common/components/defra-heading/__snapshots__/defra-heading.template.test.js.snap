// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defra-heading > renders 1`] = `
"<h1 class="govuk-visually-hidden">
  Department for Environment, Food &amp; Rural Affairs
</h1>
<div class="govuk-width-container">
  <main
    class="govuk-main-wrapper govuk-!-padding-top-4 govuk-!-padding-bottom-4 defra-banner"
  >
    <div class="govuk-grid-row">
      <div class="govuk-grid-column-one-third govuk-!-padding-left-0">
        <div class="defra-crest" data-testid="defra-heading-logo">
          <span class="defra-logo-name">
            Department<br />for Environment,<br />Food &amp; Rural Affairs
          </span>
        </div>
      </div>
      <div class="govuk-grid-column-two-thirds">
        <div class="govuk-grid-column-two-thirds govuk-!-static-padding-top-5">
          <h1 class="govuk-heading-m" data-testid="defra-heading-service-name">
            Manage rural grant applications
          </h1>
        </div>
      </div>
    </div>
  </main>
</div>
"
`;
